main.nf

```diff
@@ -0,0 +1,684 @@
+    squid -b Aligned.out.sorted.bam -c Chimeric.out.bam -o fusions
+    AnnotateSQUIDOutput.py ${gtf} fusions_sv.txt fusions_annotated.txt
+    """
+}
+
+/*************************************************************
+ * Summarizing results from tools
+ ************************************************************/
+process summary {
+    tag "$name"
+    publishDir "${params.outdir}/Report-${name}", mode: 'copy'
+ 
+    when:
+    !params.test && (params.fusioncatcher || params.star_fusion || params.ericscript || params.pizzly || params.squid)
+    
+    input:
+    set val(name), file(reads) from read_files_summary
+    file fusioncatcher from fusioncatcher_fusions.ifEmpty('')
+    file star_fusion from star_fusion_fusions.ifEmpty('')
+    file ericscript from ericscript_fusions.ifEmpty('')
+    file pizzly from pizzly_fusions.ifEmpty('')
+    file squid from squid_fusions.ifEmpty('')
+
+    output:
+    file 'fusions.txt' into summary_fusions
+    file 'summary.yaml' into summary_fusions_mq
+    file '*.html' into report
+    
+    script:
+    """
+    transformer.py -i ${fusioncatcher} -t fusioncatcher
+    transformer.py -i ${star_fusion} -t star_fusion
+    transformer.py -i ${ericscript} -t ericscript
+    transformer.py -i ${pizzly} -t pizzly
+    transformer.py -i ${squid} -t squid
+    generate_report.py fusions.txt summary.yaml -s ${name} -o .
+    """
+}
+
+/*************************************************************
+ * Visualization
+ ************************************************************/
+
+/*
+ * Fusion Inspector
+ */
+process fusion_inspector {
+    tag "$name"
+    publishDir "${params.outdir}/tools/FusionInspector", mode: 'copy'
+
+    when:
+    (params.fusion_inspector && !params.singleEnd) || (params.fusion_inspector && params.test)
+
+    input:
+    set val(name), file(reads) from read_files_fusion_inspector
+    file reference from fusion_inspector_ref.collect()
+    file summary_fusions
+
+    output:
+    file '*' into fusion_inspector_output
+
+    script:
+    """
+    FusionInspector \\
+        --fusions ${summary_fusions} \\
+        --genome_lib ${reference} \\
+        --left_fq ${reads[0]} \\
+        --right_fq ${reads[1]} \\
+        --CPU ${task.cpus} \\
+        --out_dir . \\
+        --out_prefix finspector \\
+        --prep_for_IGV
+    """
+}
+
+/*************************************************************
+ * Building report
+ ************************************************************/
+
+/*
+ * Parse software version numbers
+ */
+process get_software_versions {
+
+    when:
+    !params.test
+
+    output:
+    file 'software_versions_mqc.yaml' into software_versions_yaml
+
+    script:
+    """
+    echo $workflow.manifest.version > v_pipeline.txt
+    echo $workflow.nextflow.version > v_nextflow.txt
+    fastqc --version > v_fastqc.txt
+    multiqc --version > v_multiqc.txt
+    cat $baseDir/tools/fusioncatcher/Dockerfile | grep "VERSION" > v_fusioncatcher.txt
```

Comment:
Hmm, I see why you do it this way (multiple containers), but it's not ideal. Better to get the version number from the command directly.

Can maybe leave it like this until https://github.com/nextflow-io/nextflow/issues/879 is available though (a comment saying this would be good 👍 )

Reply:
Not all tools have a version parameter, so I went this way. It was the cleanest solution I could have found